File: src/main/java/diff/patch/accessors/FieldGetter.java
package diff.patch.accessors;

import diff.patch.Getter;

import java.lang.reflect.Field;

/**
 *
 * @author sebasjm <smarchano@primary.com.ar>
 */
public class FieldGetter<Type> implements Getter<Type> {
    final Field field;
    final Object target;

    public FieldGetter(Object target, Field field) {
        this.field = field;
        this.target = target;
        field.setAccessible(true);
    }

    @Override
    public Type get() {
        try {
            return (Type) field.get(target);
        } catch (IllegalArgumentException | IllegalAccessException ignored) {
            ignored.printStackTrace();
            return null;
        }
    }
    
}

File: src/main/java/diff/patch/accessors/FieldGetter.kt
package diff.patch.accessors

import diff.patch.Getter

import java.lang.reflect.Field

/**

 * @author sebasjm <smarchano></smarchano>@primary.com.ar>
 */
class FieldGetter<Type: Any>(internal val target: Any, internal val field: Field) : Getter<Type> {

    init {
        field.isAccessible = true
    }

    override fun get(): Type? {
        try {
            return field.get(target) as Type
        } catch (ignored: IllegalArgumentException) {
            ignored.printStackTrace()
            return null
        } catch (ignored: IllegalAccessException) {
            ignored.printStackTrace()
            return null
        }

    }

}
